packages/app/src/client/components/withSession/index.tsx

```diff
@@ -0,0 +1,261 @@
+    },
+    [dispatch]
+  );
+
+  useEffect(() => {
+    if (data?.currentUserId) {
+      handleUserLoadedSuccess(data.currentUserId);
+    }
+  }, [handleUserLoadedSuccess, data?.currentUserId]);
+
+  const reloadAppState = (appState: AppState = {}) => {
+    appStateRef.current = appState;
+  };
+
+  if (statusError) console.log("WithSession#statusError:", statusError);
+
+  if (loadingStatus) {
+    return <LoadingPage stage="status-check" />;
+  } else if (data?.currentUserId) {
+    return (
+      <Switch>
+        <Route
+          exact
+          key="/logout"
+          path="/logout"
+          component={() => (
+            <Auth0Provider
+              domain={data.auth0Config.domain}
+              clientId={data.auth0Config.clientId}
+              audience={AUTH0_AUDIENCE}
+              redirectUri={`${
+                window.location.href.split(window.location.pathname)[0]
+              }/login`}
+            >
+              <LogoutPage />
+            </Auth0Provider>
+          )}
+        />
+        <Redirect key="/login" from="/login" to={DEFAULT_PATHNAME} />
+        <Route key="app" path="*" component={() => <>{children}</>} />
+      </Switch>
+    );
+  } else {
+    return (
+      <Switch>
+        <Route
+          key="/login"
+          path="/login"
+          component={() => (
+            <Auth0Provider
+              domain={data.auth0Config.domain}
+              clientId={data.auth0Config.clientId}
+              audience={AUTH0_AUDIENCE}
+              redirectUri={`${
+                window.location.href.split(window.location.pathname)[0]
+              }/login`}
+              onRedirectCallback={reloadAppState}
+            >
+              <DetectUser
+                userLoadedSuccess={handleUserLoadedSuccess}
+                appState={appStateRef.current}
+              />
+            </Auth0Provider>
+          )}
+        />
+        <Redirect from="*" to="/login" />
+      </Switch>
+    );
+  }
+};
+
+const DetectUser = ({
+  userLoadedSuccess,
+  appState
+}: {
+  userLoadedSuccess: Function;
+  appState: AppState;
+}) => {
+  const { isLoading, isAuthenticated, loginWithRedirect } = useAuth0();
+
+  const loginHandler = useCallback(() => {
+    loginWithRedirect({
+      appState
+    });
+  }, [loginWithRedirect, appState]);
+
+  if (isLoading) return <LoadingPage stage="auth0" />;
+  else if (isAuthenticated)
+    return <CreateSession userLoadedSuccess={userLoadedSuccess} />;
+  else return <LoginPage onLogin={loginHandler} />;
+};
+
+const CreateSession = ({
+  userLoadedSuccess
+}: {
+  userLoadedSuccess: Function;
+}) => {
+  const { user, getAccessTokenSilently } = useAuth0();
+  const [accessDenied, setAccessDenied] = useState(false);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const accessToken = await getAccessTokenSilently({
+          audience: AUTH0_AUDIENCE,
+          opstraceClusterName: CLUSTER_NAME
+        });
+
```

Comment:
I see you did not carry over this comment for now: https://github.com/opstrace/opstrace/blob/afe3b291e27ad7cef24e11a1812e4b1a73372dda/packages/app/src/client/views/login/index.tsx#L161

we really should not forget about this, created a ticket here: https://github.com/opstrace/opstrace/issues/1078

Reply:
@jgehrcke shifting that comment to an actual issue is the best move so we can have a conversation about it there. 👍

Reply:
> @jgehrcke shifting that comment to an actual issue is the best move

Yes, but I would have kind of hated for it to be lost because someone thought that it can go away. Hope that makes sense, w/o any negative feelings, yeah? :)

Reply:
FYI you're making comments about an unfinished piece of work...